modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/server/persistence/RocksDbKeyValueStorage.java

```diff
@@ -833,7 +833,16 @@ protected Entry decodeEntry(byte[] key, byte[] value) {
                         return EntryImpl.empty(key);
                     }
 
-                    return doGetValue(key, targetRevision);
+                    // This is not a correct approach for using locks in terms of compaction correctness (we should block compaction for the
+                    // whole iteration duration). However, compaction is not fully implemented yet, so this lock is taken for consistency
+                    // sake. This part must be rewritten in the future.
```

Comment:
Do we have a ticket (for compaction implementation) to mention in a TODO here?

Reply:
I haven't found it =(

Reply:
Ok, it was optional